core/services/ocr2/plugins/threshold/decryption_queue.go

```diff
@@ -0,0 +1,226 @@
+package decryption_queue
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/job"
+)
+
+type CiphertextId = []byte
+
+type ThresholdDecryptor interface {
+	Decrypt(ctx context.Context, ciphertextId CiphertextId, ciphertext []byte) ([]byte, error)
+}
+
+// This interface will be replaced by the Threshold Decryption Plugin when it is ready
+type DecryptionQueuingService interface {
+	GetRequests(requestCountLimit uint32, totalBytesLimit uint32) []DecryptionRequest
+	GetCiphertext(ciphertextId CiphertextId) ([]byte, error)
+	ResultReady(ciphertextId CiphertextId, plaintext []byte)
+}
+
+type DecryptionRequest struct {
+	ciphertextId CiphertextId
+	ciphertext   []byte
+}
+
+type pendingRequest struct {
+	chPlaintext chan []byte
```

Comment:
IIUC this chan could be send-only?
```suggestion
	chPlaintext chan<- []byte
```

Reply:
And is there only one value sent? Can it be closed immediately after?

Reply:
Yes indeed.  Should I explicitly close the channel as soon as I receive the response in the `select` statement by line 85?

Reply:
The sender/owner should always close the channel. It can be closed immediately after sending the value. This does not actually interrupt anything queued in the channel buffer - if effectively queues a terminal close operation.

https://go.dev/tour/concurrency/4